hack/update-codegen.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT_RELATIVE=$(dirname "${BASH_SOURCE}")/..
+
+SCRIPT_ROOT=$(realpath "${SCRIPT_ROOT_RELATIVE}")
+
+go install k8s.io/code-generator/cmd/{client-gen,lister-gen,informer-gen,deepcopy-gen,register-gen}
+
+# Go installs the above commands to get installed in $GOBIN if defined, and $GOPATH/bin otherwise:
+GOBIN="$(go env GOBIN)"
+gobin="${GOBIN:-$(go env GOPATH)/bin}"
+
+GOROOT_TEMP="$GOROOT"
+OUTPUT_PKG="github.com/abhisheksinghbaghel/azuredisk-csi-driver/pkg/apis/client"
+APIS_PKG="github.com/abhisheksinghbaghel/azuredisk-csi-driver/pkg/apis"
+API_NAME="azuredisk"
+API_VERSION="v1beta1"
+CLIENTSET_NAME="versioned"
+CLIENTSET_PKG_NAME="clientset"
+INPUT_PKG="$APIS_PKG/$API_NAME/$API_VERSION"
+
+if [[ "${VERIFY_CODEGEN:-}" == "true" ]]; then
+  echo "Running in verification mode"
+  VERIFY_FLAG="--verify-only"
+fi
+COMMON_FLAGS="${VERIFY_FLAG:-} --go-header-file ${SCRIPT_ROOT}/hack/boilerplate/boilerplate.go.txt"
+
+echo "setting GOROOT=$GOPATH"
+export GOROOT=$GOPATH
+echo "Generating deepcopy funcs"
+"${gobin}/deepcopy-gen" --input-dirs "${INPUT_PKG}" -O zz_generated.deepcopy --bounding-dirs "${APIS_PKG}" ${COMMON_FLAGS}
```

Comment:
Rather than calling each tool individually, can you not use `vendor/k8s.io/code-generator/generate-groups.sh` as described in the article [Kubernetes Deep Dive: Code Generation for CustomResources](https://www.openshift.com/blog/kubernetes-deep-dive-code-generation-customresources)? If this is a "chicken & egg" problem, the `kubernetes / sample-controller` project uses a `[hack/tools.go](https://github.com/kubernetes/sample-controller/blob/master/hack/tools.go)` to force inclusion of `k8s.io/code-generator` in the `go.mod` file to populate the package in the `vendor` directory.

Reply:
As discussed earlier today. I was running in to an issue using the generate-groups.sh in my forked repository. I'll try to address this as a separate PR.